pkg/controller/ground.go

```diff
@@ -477,10 +482,40 @@ func (op *GroundControl) discoverOpenstackInfo(kluster *v1.Kluster) error {
 		}
 	}
 
+	if securityGroupID := copy.Spec.Openstack.SecurityGroupID; securityGroupID != "" {
+		//TODO: Validate that the securitygroup id exists
```

Comment:
forgot to add this?

Reply:
I forgot this intentionally. For this I would have needed to add yet another method to the old openstack client (`GetSecurityGroup(id string)`) which I refused to do. But maybe its better to add it now, so that when we (you) refactor the client all required method are already present.